tests/HeaderDataTest.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+declare(strict_types=1);
+
+use PHPUnit\Framework\TestCase;
+use Pantheon\EI\HeaderData;
+
+final class HeaderDataTest extends TestCase
+{
+  public function testConstructor(): void {
+    // Empty Constructor.
+    $headerData = new HeaderData();
+    $this->assertIsObject($headerData);
+
+    // Empty header data.
+    $headerData = new HeaderData([]);
+    $this->assertIsObject($headerData);
+
+    // Constructor with a small amount of array data.
+    $inputArray = [
+        'testConstructor' => 'constructor'
+    ];
+    $headerData = new HeaderData($inputArray);
+    $this->assertIsObject($headerData);
+  }
+
+  public function testGetHeader(): void {
+    $input = [
+      'HTTP_SHOULD_BE_FOUND' => 'Should be found',
+      'SERVER_NAME' => 'Entries without http_ should be ignored',
+      'HTTP_USER_AGENT' => 'Should become User_Agent'
+    ];
+    $headerData = new HeaderData($input);
+
+    $this->assertEquals($headerData->getHeader('Should-Be-Found'), 'Should be found');
+    $this->assertEmpty($headerData->getHeader('Server-Name'), 'SERVER_NAME should be ignored');
+    $this->assertEquals($headerData->getHeader('User-Agent'), 'Should become User_Agent');
+  }
+
+  public function testParseHeader(): void {
+    $input = [
+      'HTTP_AUDIENCE' => 'Parents|Children||Age:47|Name:RobLoach|Name:StevePersch|Name:AnnaMykhailova',
+      'HTTP_USER_AGENT' => 'Should just return the value',
+      'IGNORED TEST' => 'Should return an empty array',
+      'HTTP_INTEREST' => 'Carl Sagan|Richard Feynman||For Science!|With%20A Percent20',
+    ];
+    $headerData = new HeaderData($input);
+
+    // When a header doesn't exist.
+    $keyNotFound = $headerData->parseHeader('header key not found');
+    $this->assertEmpty($keyNotFound, 'Expected to return an empty array');
+    $this->assertIsArray($keyNotFound, 'Should be an array');
+
+    // Audience
+    $audience = $headerData->parseHeader('Audience');
+    $this->assertIsArray($audience['Audience']); // TODO: The Audience is nested. Intended?
```

Comment:
When the Audience element doesn't have a key, it becomes a Audience tree, which seems appropriate. Removing the TODO comment.
``` suggestion
    $this->assertIsArray($audience['Audience']);
```